terraform/aws/main.tf

```diff
@@ -0,0 +1,58 @@
+provider "aws" {
+    region = "us-east-1"
+}
+
+resource "aws_instance" "approzium" {
+    ami = var.ami
+    instance_type = var.instance_type
+    user_data = templatefile(
+        "${path.module}/scripts/install.sh.tpl",
+        {
+        download-url  = var.download-url
+        config        = file("approzium.config.yml")
+        extra-install = var.extra-install
+        logs-file     = var.logs-file
+        }
+    )
+
+    tags = {
+        Name = var.instance_name
+    }
+
+    security_groups = [aws_security_group.approzium.name]
+    key_name = var.key-name
+}
+
+
+// Security group for Approzium allows SSH, HTTP, and GRPC access
+resource "aws_security_group" "approzium" {
+    name = "approzium"
+    description = "Approzium servers"
+}
+
+resource "aws_security_group_rule" "approzium-ssh" {
+    security_group_id = aws_security_group.approzium.id
+    type = "ingress"
+    from_port = 22
+    to_port = 22
+    protocol = "tcp"
+    cidr_blocks = ["0.0.0.0/0"]
+}
+
+resource "aws_security_group_rule" "approzium-http-api" {
+    security_group_id = aws_security_group.approzium.id
+    type = "ingress"
+    from_port = 6000
+    to_port = 6000
```

Comment:
I have two questions that are totally at odds with each other! 😄 

1. Should we also do this with 6001 for GRPC?
2. Am I right in thinking that the cidr of 0.0.0.0 allows ingress from anywhere, including the public web?

I'm wondering if we should consider using a bastion server in front of the Approzium instances, and the ports would only be open to the bastion. Or some approach that by default doesn't open Approzium to the web.

Reply:
1. Totally forgot about GRPC... Nice catch!
2. you are right.
I am not familiar with a bastion server. What are the benefits of using one?